analyzer/codechecker_analyzer/cmd/fixit.py

```diff
@@ -0,0 +1,299 @@
+                             'matches any number of characters (zero or '
+                             'more). So for example "*DeadStores" will '
+                             'match "deadcode.DeadStores".')
+    parser.add_argument('--file',
+                        metavar='FILE_PATH',
+                        nargs='*',
+                        default=[],
+                        help='Filter results by file path. The file path can '
+                             'contain multiple * quantifiers which matches '
+                             'any number of characters (zero or more). So if '
+                             'you have /a/x.cpp and /a/y.cpp then "/a/*.cpp" '
+                             'selects both.')
+
+    logger.add_verbose_arguments(parser)
+    parser.set_defaults(func=main)
+
+
+def get_location_by_offset(filename, offset):
+    """
+    This function returns the line and column number in the given file which
+    is located at the given offset.
+    """
+    with open(filename, encoding='utf-8', errors='ignore') as f:
+        for row, line in enumerate(f, 1):
+            length = len(line)
+            if length < offset:
+                offset -= length
+            else:
+                return row, offset + 1
+
+
+def clang_tidy_fixit_filter(content, checker_names, file_paths, interactive,
+                            reports):
+    """
+    This function filters the content of a replacement .yaml file.
+    The function returns the set of not-existing files which to which the
+    fixits couldn't apply, and the set of changed files.
+    content -- The content of a replacement .yaml file parsed to an object by
+               yaml module.
+    checker_names -- A list of checker names possibly containing * joker
+                     characters. The full checker name must match in order to
+                     apply it.
+    file_paths -- A list of file paths to which the fixits will be applied.
+                  A file path may possibly contain * joker characters. The
+                  full path must match in order to apply it.
+    reports -- A list of CodeChecker reports. This can come from
+               "CodeChecker cmd [diff|results] ..." command in JSON format.
+    interactive -- Interactive filtering. If True then user will be asked about
+                   each fixit one by one.
+    """
+    def make_regex(parts):
+        if not parts:
+            return re.compile('.*')
+        parts = map(lambda part: re.escape(part).replace(r'\*', '.*'), parts)
+        return re.compile('|'.join(parts) + '$')
+
+    def ask_user(item):
+        print(yaml.dump(item))
+        prompt = "y/<Enter> - Yes\nOther     - No\nCtrl-C    - Cancel\nApply: "
+        return input(prompt) in "Yy"
+
+    def exists_collect(file_path):
+        if not os.path.isfile(file_path):
+            not_existing_files.add(file_path)
+            return False
+        return True
+
+    if reports:
```

Comment:
Is it a good idea to define `match_reports` as a function if there are some reports otherwise define it as the bool function?

Wouldn't it be better to handle it in the `match_reports` function like so:
```
def match_reports(diag_msg):
    """ TODO """
    if not reports:
        return bool(diag_msg)
    ....
```

Reply:
I think this is a great idea using `bool` constructor as a functor, it's a pity that you don't like it :smile: This way you don't need to check `reports` every time, but ok, I changed it :smile: